playbooks/robusta_playbooks/persistent_volume_actions.py

```diff
@@ -0,0 +1,170 @@
+    pv = event.get_persistentvolume()
+    pv_claimref = pv.spec.claimRef
+
+    try:
+
+        if pv_claimref != None:
+            # Do this if there is a PVC attached to PV
+            pvc_obj = PersistentVolumeClaim.readNamespacedPersistentVolumeClaim(
+                name=pv_claimref.name, namespace=pv_claimref.namespace).obj
+            pod_obj = get_pod_related_to_pvc(pvc_obj, pv)
+            if (pod_obj != None):
+                # Do this if a Pod is using PVC
+
+                pod = RobustaPod.find_pod(
+                    pod_obj.metadata.name, pod_obj.metadata.namespace)
+                volume_mount_name = None
+
+                # Find name of the mounted volume on pod
+                for volume in pod.spec.volumes:
+                    if volume.persistentVolumeClaim.claimName == pv_claimref.name:
+                        volume_mount_name = volume.name
+                        break
+
+                # Use name of the mounted volume to find the correct volume mount
+                container_found_flag = False
+                container_volume_mount = None
+                for container in pod.spec.containers:
+                    if (container_found_flag):
+                        break
+                    if(container.volumeMounts):
+                        for volume_mount in container.volumeMounts:
+                            if volume_mount_name == volume_mount.name:
+                                container_volume_mount = volume_mount
+                                container_found_flag = True
+                                break
+                
+                result = pod.exec(
+                    f"ls -R {container_volume_mount.mountPath}/")
+                finding.title = f"Files present on persistent volume {pv.metadata.name} are: "
+                finding.add_enrichment(
+                    [
+
+                        FileBlock(
+                            f"Data.txt: ", result.encode()),
+
+                    ]
+                )
+            else:
+                # Do this if no Pod is attached to PVC
+                reader_pod = persistent_volume_reader(persistent_volume=pv)
+                result = reader_pod.exec(
+                    f"ls -R {reader_pod.spec.containers[0].volumeMounts[0].mountPath}")
+                finding.title = f"Files present on persistent volume {pv.metadata.name} are: "
+                finding.add_enrichment(
+                    [
+
+                        FileBlock(
+                            f"Data.txt: ", result.encode()),
+
+                    ]
+                )
+                # delete the reader pod
+                reader_pod.delete()
+
+        else:
+            finding.add_enrichment(
+                [
+                    MarkdownBlock(
+                        f"ERROR: Persistent volume named {pv.metadata.name} have no persistent volume claim."
+                    ),
+                ]
+            )
+
+    except RuntimeError as e:
```

Comment:
Usually, when there's an exception, we don't send it to the user.
Most user will not know what to do with it.

In the framework, we have code that except any exception thrown from an action, and adds some "user friendly" text.
So I think you can remove the except sections here

Reply:
Makes sense. 
Should I just remove the exceptions completely or log the exceptions here and not send them to sinks instead which can help easier debugging of action in future.

Reply:
Please take a look at this line:
https://github.com/robusta-dev/robusta/blob/5a404cbde06ff5635d0e3fc20e4b295dfb5310b1/src/robusta/core/playbooks/playbooks_event_handler_impl.py#L229

We catch and print any exception thrown from the action, and add an enrichment saying: "Oops... Error processing {action.action_name}"

So the exception is already logged, no need to catch and print it